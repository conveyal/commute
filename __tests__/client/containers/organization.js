/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {
  expectDeleteAnalysis,
  expectDeleteGroup,
  expectDeleteOrganization,
  expectDeleteSite
} from '../../test-utils/actions'
import {makeMockStore, mockStores} from '../../test-utils/mock-data.js'

import Organization from '../../../client/containers/organization'

describe('Container > Organization', () => {
  it('Organization View loads', () => {
    // mount component
    const tree = mount(
      <Provider store={makeMockStore(mockStores.complexOrganization)}>
        <Organization
          params={{organizationId: '2'}}
          />
      </Provider>
    )
    expect(mountToJson(tree.find(Organization))).toMatchSnapshot()
  })

  it('Delete Organization', () => {
    const mockStore = makeMockStore(mockStores.oneSimpleOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the organization view
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Organization
          params={{organizationId: '1'}}
          />
      </Provider>
    )

    // When the user clicks the delete button for an existing organization
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('.btn-group').first().find('button').last()
    deleteButton.simulate('click')

    expectDeleteOrganization(mockStore.getActions())
  })

  it('Delete Site', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the organization view
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Organization
          params={{organizationId: '2'}}
          />
      </Provider>
    )

    // When the user clicks the delete button for an existing site
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('table').at(1).find('button').last()
    deleteButton.simulate('click')

    expectDeleteSite(mockStore.getActions())
  })

  it('Delete Commuter Group', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the organization view
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Organization
          params={{organizationId: '2'}}
          />
      </Provider>
    )

    // When the user clicks the delete button for an existing commuter group
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('table').at(3).find('button').last()
    deleteButton.simulate('click')

    expectDeleteGroup(mockStore.getActions())
  })

  it('Delete Analysis', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the organizations view
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Organization
          params={{organizationId: '2'}}
          />
      </Provider>
    )

    // When the user clicks the delete button for an existing analysis
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('table').at(5).find('button').last()
    deleteButton.simulate('click')

    expectDeleteAnalysis(mockStore.getActions())
  })
})
