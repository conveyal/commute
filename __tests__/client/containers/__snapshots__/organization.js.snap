exports[`Container > Organization Delete Analysis 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organization/organization-2",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > Organization Delete Commuter Group 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organization/organization-2",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > Organization Delete Organization 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/agency/agency-3",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > Organization Delete Site 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organization/organization-2",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > Organization Organization View loads 1`] = `
"<div class=\"container\">
  <div class=\"row\">
    <div class=\"col-xs-12\">
      <h2>
        <i class=\"fa fa-shield fa-fw \"></i>

        <!-- react-text: 6 -->Mock Organization<!-- /react-text -->
      </h2>
      <div class=\"btn-group\">
        <button to=\"/organization/organization-2/edit\" type=\"button\" class=\"btn btn-warning\">Edit</button>
        <button type=\"button\" class=\"btn btn-danger\">Delete</button>
      </div>
      <p>Below are this organization\'s sites, commuter groups and analyses.</p>
      <h3>

        <!-- react-text: 12 -->Sites<!-- /react-text -->
        <button to=\"/organization/organization-2/site/create\" type=\"button\" class=\"pull-right btn btn-default\"><span>Create a new site</span> 
          <i class=\"fa fa-building fa-fw \"></i>
        </button>
      </h3>
      <p>A site is a location of a building or new address that you want to use as the centerpoint of your commutes.</p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none; min-width: NaNpx;\">
                  <col class=\"\" style=\"min-width: NaNpx;\">
                    <col class=\"\" style=\"min-width: NaNpx;\">
                      <col class=\"\" style=\"min-width: NaNpx;\">
              </colgroup>
              <thead>
                <tr>
                  <th class=\"\" data-is-only-head=\"false\" style=\"text-align: left; display: none;\" data-field=\"_id\">
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Name\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 31 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Address\" style=\"text-align: left;\" data-field=\"address\">

                    <!-- react-text: 34 -->Address<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Tools\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 37 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td class=\"\" style=\"text-align: left; display: none;\">site-2</td>
                  <td class=\"\" style=\"text-align: left;\">
                    <a>Acme Corp</a> 
                  </td>
                  <td class=\"\" style=\"text-align: left;\">123 ABC St</td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div class=\"btn-group\">
                      <button to=\"/site/site-2/edit\" type=\"button\" class=\"btn btn-warning\">Edit</button>
                      <button type=\"button\" class=\"btn btn-danger\">Delete</button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <h3>

        <!-- react-text: 57 -->Commuter Groups<!-- /react-text -->
        <button to=\"/organization/organization-2/group/create\" type=\"button\" class=\"pull-right btn btn-default\"><span>Create a new group</span> 
          <i class=\"fa fa-users fa-fw \"></i>
        </button>
      </h3>
      <p>A commuter group is a list of commuters that can commute to a particular site.</p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none; min-width: NaNpx;\">
                  <col class=\"\" style=\"min-width: NaNpx;\">
                    <col class=\"\" style=\"min-width: NaNpx;\">
                      <col class=\"\" style=\"min-width: NaNpx;\">
              </colgroup>
              <thead>
                <tr>
                  <th class=\"\" data-is-only-head=\"false\" style=\"text-align: left; display: none;\" data-field=\"_id\">
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Name\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 76 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Commuters\" style=\"text-align: left;\" data-field=\"commuters\">

                    <!-- react-text: 79 -->Commuters<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Tools\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 82 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td class=\"\" style=\"text-align: left; display: none;\">group-2</td>
                  <td class=\"\" style=\"text-align: left;\">
                    <a>Mock Group</a> 
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div>1</div>
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <button type=\"button\" class=\"btn btn-danger\">Delete</button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <h3>

        <!-- react-text: 101 -->Analyses<!-- /react-text -->
        <button to=\"/organization/organization-2/analysis/create\" type=\"button\" class=\"pull-right btn btn-default\"><span>Create a new analysis</span> 
          <i class=\"fa fa-bar-chart fa-fw \"></i>
        </button>
      </h3>
      <p>An analysis calculates commuting statistics for a pairing of a commuter group and site.</p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none; min-width: NaNpx;\">
                  <col class=\"\" style=\"min-width: NaNpx;\">
                    <col class=\"\" style=\"min-width: NaNpx;\">
              </colgroup>
              <thead>
                <tr>
                  <th class=\"\" data-is-only-head=\"false\" style=\"text-align: left; display: none;\" data-field=\"_id\">
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Name\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 119 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Tools\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 122 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td class=\"\" style=\"text-align: left; display: none;\">analysis-2</td>
                  <td class=\"\" style=\"text-align: left;\">
                    <a>An Analysis</a> 
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <button type=\"button\" class=\"btn btn-danger\">Delete</button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
